Yank/yank.py

```diff
@@ -245,6 +251,237 @@ def get_region(self, region_name):
         # Return a copy to ensure people cant tweak the region outside of the api
         return copy.copy(self._regions[region_name])
 
+    def select(self, selection, as_set=False, sort_by='auto') -> Union[List[int], Set[int]]:
+        """
+        Select atoms based on selection format which can be a number of formats:
+
+        * Single integer (a bit redundant)
+        * Iterable of integer (also a bit redundant)
+        * Complex Region selection
+        * MDTraj String
+        * **Future Feature** SMARTS Selection
+
+        This method will never return duplicate atom numbers.
+
+        The ``sort_by`` method controls how the output is sorted before being returned, see the details in the
+        Parameters block for information about each option
+
+        The Complex Region string returns an the set of atoms derived from the arguments using logical operators
+        ``and`` and ``or``
+        along with grouping through parenthesis . For example, assume you
+        have two regions ``regionA = [0,1,2,3]`` and ``regionB = [2,3,4,5]``. You can do operations such as the
+        following:
+
+             ``regionA and regionB`` yields ``[2,3]``, which is the intersection of the regions.
+
+             ``regionA or regionB`` yields ``[0,1,2,3,4,5]``, which is the union of the regions.
+
+        More complex statements with more regions will also work, and statements can be grouped with ``()``.
+
+        Parameters
+        ----------
+        selection : str or
+            String defining the selection
+        as_set : bool, Default False
+            Determines output format. Returns a Set if True, otherwise output is a list.
+        sort_by : str or None, Default: 'auto'
+            Determine how to sort the output if ``as_set`` is False.
+
+            * 'auto': Let the selection string determine how to sort it out based on its priorities
+            * 'ordered': Atoms are sorted index, smallest to largest
+            * 'inferred_by_region': Atoms are sorted by which region in the provided ``selection_string`` occurs first.
```

Comment:
I would call these two something like `sorty_by='index'` and `sorty_by='region_order'`.